pkgs/test/lib/src/runner/browser/dom.dart

```diff
@@ -2,29 +2,74 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'dart:js_interop';
 import 'dart:js_util' as js_util;
 
 import 'package:js/js.dart';
 
+// Conversion functions to help with the migration.
+extension ObjectToJSAnyExtension on Object? {
+  JSAny? get toJSAnyShallow {
+    if (const bool.fromEnvironment('dart.library.html')) {
+      // Cast is necessary on Wasm backends and in the future on JS backends as
+      // well.
+      // ignore: unnecessary_cast
+      return this as JSAny?;
+    } else {
+      return toJSAnyDeep;
```

Comment:
this seems a bit weird for a function called `toJsAnyShallow` haha, lets at least put a comment on the function mentioning that it might actually be deep.